vsintegration/src/FSharp.Editor/CodeFix/AddOpenCodeFixProvider.fs

```diff
@@ -148,7 +148,7 @@ type internal FSharpAddOpenCodeFixProvider
         asyncMaybe {
             let! options = projectInfoManager.TryGetOptionsForEditingDocumentOrProject context.Document
             let! sourceText = context.Document.GetTextAsync(context.CancellationToken)
-            let! parsedInput, checkResults = checker.ParseAndCheckDocument(context.Document, options, sourceText)
+            let! _, parsedInput, checkResults = checker.ParseAndCheckDocument(context.Document, options, sourceText)
```

Comment:
Yes, this scenario is possible I think, but it's OK. Not OK is the initial file checking where all the features call `ParseAndCheckFileInProject` at the same time, got `None` for stale results and multiple checks start simultaniously, increasing check time by N times :( We need to set a flag that a file is being checked and callers of `ParseAndCheckFileInProject` should be anynchronously blocked until results are available and written to the cache. Maybe create a version of `TryGetRecentCheckResultsForFile` that blocks until check results appears in the cache?